File: shared/contracts/ts/v1/note.ts
export interface SaveNoteRequest {
  note_id: string;
  note_title: string;
  subject_id: string;
  tags: string[];
  is_pinned: boolean;
  is_archived: boolean;
  content_json: Record<string, unknown>;
  content_text: string;
  updated_at: string;
}

export interface SaveNoteResponse {
  note_id: string;
  saved_at: string;
  version: number;
}

export interface GetNoteResponse {
  note_id: string;
  note_title: string;
  subject_id: string;
  tags: string[];
  is_pinned: boolean;
  is_archived: boolean;
  content_json: Record<string, unknown>;
  content_text: string;
  updated_at: string;
  version: number;
}

export interface NoteSummary {
  note_id: string;
  note_title: string;
  subject_id: string;
  tags: string[];
  is_pinned: boolean;
  is_archived: boolean;
  content_text: string;
  updated_at: string;
  version: number;
}

export interface ListNotesResponse {
  items: NoteSummary[];
  total: number;
}

File: shared/contracts/python/v1/note.py
from __future__ import annotations

from pydantic import BaseModel, Field


class SaveNoteRequest(BaseModel):
    note_id: str = Field(min_length=1)
    content_json: dict[str, object] = Field(min_length=1)
    content_text: str = Field(min_length=1)
    updated_at: str = Field(min_length=1)


class SaveNoteResponse(BaseModel):
    note_id: str = Field(min_length=1)
    saved_at: str = Field(min_length=1)
    version: int = Field(ge=1)


class GetNoteResponse(BaseModel):
    note_id: str = Field(min_length=1)
    content_json: dict[str, object] = Field(min_length=1)
    content_text: str = Field(min_length=1)
    updated_at: str = Field(min_length=1)
    version: int = Field(ge=1)


class NoteSummary(BaseModel):
    note_id: str = Field(min_length=1)
    content_text: str = Field(min_length=1)
    updated_at: str = Field(min_length=1)
    version: int = Field(ge=1)


class ListNotesResponse(BaseModel):
    items: list[NoteSummary]
    total: int = Field(ge=0)
